storage/engine/mvcc.go

```diff
@@ -627,25 +627,24 @@ func mvccPutInternal(engine Engine, ms *proto.MVCCStats, key proto.Key, timestam
 		return util.Errorf("key %q value contains both a byte slice and an integer value: %+v", key, value)
 	}
 
-	meta := &buf.meta
 	metaKey := mvccEncodeKey(buf.key[0:0], key)
-	ok, origMetaKeySize, origMetaValSize, err := engine.GetProto(metaKey, meta)
+	ok, origMetaKeySize, origMetaValSize, err := engine.GetProto(metaKey, &buf.meta)
 	if err != nil {
 		return err
 	}
-	origAgeSeconds := timestamp.WallTime/1E9 - meta.Timestamp.WallTime/1E9
 
 	// Verify we're not mixing inline and non-inline values.
 	putIsInline := timestamp.Equal(proto.ZeroTimestamp)
-	if ok && putIsInline != meta.IsInline() {
+	if ok && putIsInline != buf.meta.IsInline() {
 		return util.Errorf("%q: put is inline=%t, but existing value is inline=%t",
-			metaKey, putIsInline, meta.IsInline())
+			metaKey, putIsInline, buf.meta.IsInline())
 	}
 	if putIsInline {
 		var metaKeySize, metaValSize int64
 		if value.Deleted {
 			metaKeySize, metaValSize, err = 0, 0, engine.Clear(metaKey)
 		} else {
+			meta := &proto.MVCCMetadata{}
```

Comment:
Ah, nevermind. `meta` doesn't exist at this point. The old code's intention was to avoid an allocation. You could do so here by using `buf.meta`. It's just scratch space.